components/script/dom/writablestreamdefaultcontroller.rs

```diff
@@ -322,27 +340,40 @@ impl WritableStreamDefaultController {
 
     /// <https://streams.spec.whatwg.org/#writable-stream-default-controller-clear-algorithms>
     fn clear_algorithms(&self) {
-        // Set controller.[[writeAlgorithm]] to undefined.
-        self.write.borrow_mut().take();
-
-        // Set controller.[[closeAlgorithm]] to undefined.
-        self.close.borrow_mut().take();
-
-        // Set controller.[[abortAlgorithm]] to undefined.
-        self.abort.borrow_mut().take();
-
+        if let Some(underlying_sink_type) = self.underlying_sink_type.borrow().as_ref() {
+            match underlying_sink_type {
+                UnderlyingSinkType::Js {
+                    abort,
+                    start: _,
+                    close,
+                    write,
+                } => {
+                    // Set controller.[[writeAlgorithm]] to undefined.
+                    write.borrow_mut().take();
+
+                    // Set controller.[[closeAlgorithm]] to undefined.
+                    close.borrow_mut().take();
+
+                    // Set controller.[[abortAlgorithm]] to undefined.
+                    abort.borrow_mut().take();
+                },
+                UnderlyingSinkType::Transfer => todo!(),
+                UnderlyingSinkType::Transform(_) => {
+                    *self.underlying_sink_type.borrow_mut() = None;
+                },
+            }
+        }
         // Set controller.[[strategySizeAlgorithm]] to undefined.
         self.strategy_size.borrow_mut().take();
     }
 
-    /// <https://streams.spec.whatwg.org/#set-up-writable-stream-default-controllerr>
+    /// <https://streams.spec.whatwg.org/#set-up-writable-stream-default-controller>
     #[allow(unsafe_code)]
     pub(crate) fn setup(
         &self,
         cx: SafeJSContext,
         global: &GlobalScope,
         stream: &WritableStream,
-        start: &Option<Rc<UnderlyingSinkStartCallback>>,
```

Comment:
As stated above, it was better to pass the algo here, and then forget about it, because this is the only place where it is used.